src/main/java/com/iexec/core/replicate/ReplicatesService.java

```diff
@@ -418,11 +414,50 @@ public void updateReplicateStatus(OptimisticLockingFailureException exception,
      *   3) if worker did succeed onChain when CONTRIBUTED/REVEALED.
      *   4) if worker did upload when RESULT_UPLOADING.
      */
+
+    /**
+     * This method updates a replicate while caring about thread safety.
+     * A single replicate can then NOT be updated twice at the same time.
+     * This method should be preferred to
+     * {@link ReplicatesService#updateReplicateStatusWithoutThreadSafety(String, String, ReplicateStatusUpdate, UpdateReplicateStatusArgs)}!
+     *
+     * @param chainTaskId Chain task id of the task whose replicate should be updated.
+     * @param walletAddress Wallet address of the worker whose replicate should be updated.
+     * @param statusUpdate Info about the status update - new status, date of update, ...
+     * @param updateReplicateStatusArgs Optional args used to update the status.
+     * @return An optional next action for the worker.
+     */
     @Retryable(value = {OptimisticLockingFailureException.class}, maxAttempts = 100)
     public Optional<TaskNotificationType> updateReplicateStatus(String chainTaskId,
```

Comment:
Just to understand: calls on locks are non-blocking and we try a bunch of times until the lock is obtained ? (A try every second for 100 seconds)
When the deadline is reached, we go to the @Recover annotated method.

Reply:
Calls on locks are blocking.
The `@Retryable` annotation helps with database issues - the database mechanism also uses some locking feature. So we try 100 times if the database fails to save the new replicate status.